osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -274,8 +276,37 @@ private partial class ModSettingsEnumDropdown<T> : SettingsEnumDropdown<T>
 
             private partial class ModDropdownControl : DropdownControl
             {
-                // Set menu's max height low enough to workaround nested scroll issues (see https://github.com/ppy/osu-framework/issues/4536).
-                protected override DropdownMenu CreateMenu() => base.CreateMenu().With(m => m.MaxHeight = 100);
+                protected override DropdownMenu CreateMenu() => new ModDropdownMenu();
+
+                private partial class ModDropdownMenu : OsuDropdownMenu
+                {
+                    public ModDropdownMenu()
+                    {
+                        // Set menu's max height low enough to workaround nested scroll issues (see https://github.com/ppy/osu-framework/issues/4536).
+                        MaxHeight = 100;
+                    }
+
+                    protected override void UpdateSize(Vector2 newSize)
+                    {
+                        base.UpdateSize(newSize);
+
+                        // todo: probably move this to OsuDropdown so that settings overlay can benefit from this as well.
+                        if (newSize.Y > 0)
+                        {
+                            var scroll = this.FindClosestParent<OsuScrollContainer>();
+
+                            if (scroll != null)
+                            {
+                                const float padding = 15;
+
+                                float target = scroll.GetChildPosInContent(this, new Vector2(0, newSize.Y + padding));
+
+                                if (target > scroll.Current + scroll.DisplayableContent)
+                                    scroll.ScrollTo(target - scroll.DisplayableContent);
```

Comment:
While we're on this I would like to say that I find that the current implementation of this, which smushes this behaviour into `SettingSourceAttribute`, is very very very *very **very*** bad. It is a situational interaction which got put into a component that is *not* mod setting specific anymore.

To that end I would propose the following diff over the current code, which addresses both the code quality and the behaviour concerns simultaneously:

```diff
diff --git a/osu.Game/Configuration/SettingSourceAttribute.cs b/osu.Game/Configuration/SettingSourceAttribute.cs
index f87c8de0cb..1e425c88a6 100644
--- a/osu.Game/Configuration/SettingSourceAttribute.cs
+++ b/osu.Game/Configuration/SettingSourceAttribute.cs
@@ -13,10 +13,8 @@
 using osu.Framework.Extensions.TypeExtensions;
 using osu.Framework.Graphics;
 using osu.Framework.Localisation;
-using osu.Game.Graphics.Containers;
 using osu.Game.Graphics.UserInterface;
 using osu.Game.Overlays.Settings;
-using osuTK;
 
 namespace osu.Game.Configuration
 {
@@ -276,37 +274,8 @@ private partial class ModSettingsEnumDropdown<T> : SettingsEnumDropdown<T>
 
             private partial class ModDropdownControl : DropdownControl
             {
-                protected override DropdownMenu CreateMenu() => new ModDropdownMenu();
-
-                private partial class ModDropdownMenu : OsuDropdownMenu
-                {
-                    public ModDropdownMenu()
-                    {
-                        // Set menu's max height low enough to workaround nested scroll issues (see https://github.com/ppy/osu-framework/issues/4536).
-                        MaxHeight = 100;
-                    }
-
-                    protected override void UpdateSize(Vector2 newSize)
-                    {
-                        base.UpdateSize(newSize);
-
-                        // todo: probably move this to OsuDropdown so that settings overlay can benefit from this as well.
-                        if (newSize.Y > 0)
-                        {
-                            var scroll = this.FindClosestParent<OsuScrollContainer>();
-
-                            if (scroll != null)
-                            {
-                                const float padding = 15;
-
-                                float target = scroll.GetChildPosInContent(this, new Vector2(0, newSize.Y + padding));
-
-                                if (target > scroll.Current + scroll.DisplayableContent)
-                                    scroll.ScrollTo(target - scroll.DisplayableContent);
-                            }
-                        }
-                    }
-                }
+                // Set menu's max height low enough to workaround nested scroll issues (see https://github.com/ppy/osu-framework/issues/4536).
+                protected override DropdownMenu CreateMenu() => base.CreateMenu().With(m => m.MaxHeight = 100);
             }
         }
     }
diff --git a/osu.Game/Overlays/Mods/ModCustomisationPanel.cs b/osu.Game/Overlays/Mods/ModCustomisationPanel.cs
index 240569031c..ef0429f43c 100644
--- a/osu.Game/Overlays/Mods/ModCustomisationPanel.cs
+++ b/osu.Game/Overlays/Mods/ModCustomisationPanel.cs
@@ -10,8 +10,10 @@
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Effects;
 using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
 using osu.Framework.Input.Bindings;
 using osu.Framework.Input.Events;
+using osu.Framework.Testing;
 using osu.Game.Configuration;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Containers;
@@ -193,6 +195,9 @@ protected override void Update()
         {
             base.Update();
             scrollContainer.Height = Math.Min(scrollContainer.AvailableContent, DrawHeight - header_height);
+
+            if (this.ChildrenOfType<Menu>().FirstOrDefault(m => m.State == MenuState.Open) is Menu openMenu)
+                scrollContainer.ScrollIntoView(openMenu, extraScroll: 30);
         }
 
         private partial class FocusGrabbingContainer : InputBlockingContainer

```

Reply:
The entire feature been removed as per a discussion in [Discord](https://discord.com/channels/188630481301012481/188630652340404224/1258271932319531030).